docs/recipes/travis_build_stages.md

```diff
@@ -0,0 +1,90 @@
+# Using semantic-release with [Travis build stages](https://docs.travis-ci.com/user/build-stages/)
+
+![Travis Build Stages screenshot](https://cloud.githubusercontent.com/assets/3729517/25229553/0868909c-25d1-11e7-9263-b076fdef9288.gif)
+
+Build stages are a great way to run semantic release compared to running it in `after_success`:
+
+* The release stage can use Node 8, while the other stages can use other versions, even use a completely different language.
+* It allows you to easily release a project that is not even written in JavaScript, without impacting the job configuration of your test jobs in any way.
+* Instead of having to figure out a "build leader" to do the release and poll the Travis API to wait until all jobs passed, Travis can handle only running the release step after the other jobs passed.
+* If the release failed, you can see at a glance that only the release stage failed while your test stage succeded.
+* Travis can take care of not running the whole release stage at all on non-master branches and PRs.
+
+### Annotated example `.travis.yml`
+
+```yml
+# This could even be a different language
+language: node_js
+
+# Define any version matrix you want for testing
+node_js:
+  - '8'
+  - '6'
+
+# This is inherited by all jobs
+cache:
+  directories:
+    - ~/.npm
+
+# This will be inherited in the release stage
+install:
+  - npm install
+
+# This is the script that will only run for your test stage
+script:
+  - npm run build
+  - npm run lint
+  - npm test
```

Comment:
I suggest to leave out the `script` part altogether, I think the receipt should focus on the semantic-release, weather you have a build or lint script that you want to run next to `npm test` depends on the respective project

Reply:
I wanted to highlight that you must not forget to run build again in the release step if you have a build process. I've forgot to do that a couple of times and ended up releasing uncompiled packages

Reply:
Per @gr2m comment